### PR TITLE
Update bands workchain section

### PR DESCRIPTION
**Pull request by borellim**
This is the continuation of PR #24, and should address the remaining change requests.

In addition to what Daniele did already:
* I rebased the branch;
* I fixed some broken references;
* I moved the notes on appendices being optional, from the bottom of the last section to the top of each appendix, because the text is now split in different pages;
* I slightly tweaked the text again.

Question / note:
* refs and numrefs in the middle of italic text do not seem to work.

### CODE REVIEW

**Comment by ltalirz** [comment]
@borellim I'll check, in the meanwhile can you fix the conflict?

File: docs/pages/2019_MARVEL_Psik_MaX/sections/appendix_input_validation.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+*These appendices consist of optional exercises, and are mentioned
```
Comment (ltalirz) [inline docs/pages/2019_MARVEL_Psik_MaX/sections/appendix_input_validation.md:1]: why not move this to the top-level index.rst?

File: docs/pages/2019_MARVEL_Psik_MaX/sections/bands.rst [hunk 2]
```diff
@@ -1,46 +1,55 @@
+.. _bands:
+
 A real-world WorkChain: computing a band structure
 ==================================================
 
-**Note**: *If you still have enough time, you might want to check first
-Appendix [sec:convpressure] before continuing with this section.*
+**Note:** *If you still have enough time, you might want to check first*
```
Comment (ltalirz) [inline docs/pages/2019_MARVEL_Psik_MaX/sections/bands.rst]: by the way, RST has the much nicer

.. note::

   ...

File: docs/pages/2019_MARVEL_Psik_MaX/sections/appendix_input_validation.md (hunk 1 above)
Comment (borellim) [reply docs/pages/2019_MARVEL_Psik_MaX/sections/appendix_input_validation.md:1]: Because people will probably not see it...? :) As you prefer though.

Comment (borellim) [reply docs/pages/2019_MARVEL_Psik_MaX/sections/appendix_input_validation.md:1]: I'd rather keep them here. They are redundant in the index; here instead they serve as a reminder for people to not focus on these parts unless they have understood the rest or they have plenty of time. The appendix section titles do not say "Appendix 1.", just "1.".

Comment (ltalirz) [reply docs/pages/2019_MARVEL_Psik_MaX/sections/appendix_input_validation.md:1]: Ok, fine

File: docs/pages/2019_MARVEL_Psik_MaX/sections/bands.rst (hunk 2 above)
Comment (borellim) [reply docs/pages/2019_MARVEL_Psik_MaX/sections/bands.rst]: Thanks, done.

**Comment by borellim** [comment]
I rebased to fix the conflicts (and the tests, hopefully), taking the occasion to apply your changes and squash some commits.

**Comment by ltalirz** [comment]
Thanks! 
You can remove the png files, they're already in images/bandstructures (which are the ones you're using)

**Comment by borellim** [comment]
Whoops.... Thanks for noticing. Is this ok now?

**Comment by ltalirz** [comment]
Looks good, thanks!

**Comment by d-tomerini** [comment]
Thanks a lot for fixing everything, @borellim !

**Pull request merged**